### PR TITLE
Adds distress beacons instead of the current ERT call

### PR DESCRIPTION
**Pull request by NonQueueingMatt**
This is my message to my master:
This is a fight you cannot win,
I think that past your great disasters
Their victory stirs below your skin
If you proceed I will not blame you
I will move forth and win your war!

--

Adds distress beacons with a chance-based roll system.

TO-DO:

- [x] Add ghost spawner for the ERT type.

- [x] Add the shuttle for the ERT.

- [x] Dynamic gear spawning.

- [x] Leader preference + support.

- [x] TCFL datum

- [x] NT-ERT datum

- [x] Mercenary (neutral ERT) datum

### CODE REVIEW

**Comment by XanderDox** [comment]
We do not need a syndicate team showing up in the middle of X round type or even extended. This is a bad PR.

**Comment by LadyFowls** [comment]
> We do not need a syndicate team showing up in the middle of X round type or even extended. This is a bad PR.

Why is it a bad pr?

File: code/datums/outfits/ert/ert.dm [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+		/obj/item/ammo_magazine/a556,
+		/obj/item/ammo_magazine/a556/ap
+	)
+	
+/datum/outfit/admin/ert/nanotrasen/specialist
+	belt = null
+	belt_contents = null
+
+/datum/outfit/admin/ert/nanotrasen/specialist/post_equip(mob/living/carbon/human/H)
+	switch(alert("Do you want to be an engineering or medical specialist?", "Response Team", "Engineering", "Medical")
```
Comment (Arrow768) [inline code/datums/outfits/ert/ert.dm]: That is not a good way of doing that.
Have different ghostroles for the different specialists.

Comment (NonQueueingMatt) [reply code/datums/outfits/ert/ert.dm]: Got it

Comment (NonQueueingMatt) [reply code/datums/outfits/ert/ert.dm]: Solved

File: code/controllers/subsystems/responseteam.dm [hunk 2]
```diff
@@ -2,82 +2,59 @@
...
-			percentage_antagonists = 0
-			percentage_dead = 0
-		else
-			percentage_antagonists = round(100 * antagonists / total)
-			percentage_dead = round(100 * deadcount / total)
-		
-
-		switch(get_security_level())
-			if("green")
-				progression_chance += config.ert_green_inc
```
Comment (Arrow768) [inline code/controllers/subsystems/responseteam.dm]: Make sure to remove the unneeded config values from the config system and the example config file

File: code/controllers/subsystems/responseteam.dm [hunk 3]
```diff
@@ -2,82 +2,59 @@
...
-/datum/controller/subsystem/responseteam/proc/trigger_armed_response_team(var/forced_choice = FALSE)
+/datum/controller/subsystem/responseteam/proc/pick_random_team()
+	var/datum/responseteam/result
+	var/probability = rand(1, 100)
+	var/tally = 0
+	for(var/datum/responseteam/ert in available_teams) //We need a loop to keep going through each candidate to be sure we find a good result.
+		if((ert.chance + tally) <= probability) //Check every available ERT's chance. Keep going until we add enough to the tally so that we have a certain result.
+			tally += ert.chance
+			continue
+			result = ert
```
Comment (Arrow768) [inline code/controllers/subsystems/responseteam.dm]: Improper indentation. 
Won't be executed as it's located after the xontinue

File: code/game/antagonist/outsider/ert.dm [hunk 4]
```diff
@@ -27,20 +27,4 @@ var/datum/antagonist/ert/ert
 	..()
 	ert = src
 
-/datum/antagonist/ert/greet(var/datum/mind/player)
-	if(!..())
-		return
-	to_chat(player.current, "The Emergency Response Team works for Asset Protection; your job is to protect [current_map.company_name]'s assets. There is a code red alert on [station_name()], you are tasked to go and fix the problem.")
-	to_chat(player.current, "You should first gear up and discuss a plan with your team. More members may be joining, don't move out before you're ready.")
```
Comment (JohnWildkins) [inline code/game/antagonist/outsider/ert.dm]: I'm assuming these will be handled on a per-team basis instead, because this is useful albeit basic info

Comment (NonQueueingMatt) [reply code/game/antagonist/outsider/ert.dm]: Yeah, ghost spawners have a custom spawn message that you can set.

File: code/controllers/subsystems/responseteam.dm (hunk 2 above)
Comment (NonQueueingMatt) [reply code/controllers/subsystems/responseteam.dm]: Solved I think

**Comment by NonQueueingMatt** [comment]
!wip

**Comment by NonQueueingMatt** [comment]
!review

File: code/controllers/subsystems/ghostroles.dm [hunk 5]
```diff
@@ -159,7 +159,7 @@
 			S.enable()
 			to_chat(usr, "Ghost spawner enabled: [S.name]")
 			SSvueui.check_uis_for_change(src) //Update all the UIs to update the status of the spawner
-			for(var/i in S.spawnpoints)
+			for(var/i in spawnpoints)
```
Comment (NonQueueingMatt) [inline code/controllers/subsystems/ghostroles.dm]:
```suggestion
			for(var/i in S.spawnpoints)
```

File: code/controllers/subsystems/ghostroles.dm [hunk 6]
```diff
@@ -171,6 +171,6 @@
 			S.disable()
 			to_chat(usr, "Ghost spawner disabled: [S.name]")
 			SSvueui.check_uis_for_change(src) //Update all the UIs to update the status of the spawner
-			for(var/i in S.spawnpoints)
+			for(var/i in spawnpoints)
```
Comment (NonQueueingMatt) [inline code/controllers/subsystems/ghostroles.dm]:
```suggestion
			for(var/i in S.spawnpoints)
```

File: html/changelogs/mattatlas-mybloodpacttoskull.yml [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+# INDENT WITH TWO SPACES.  NOT TABS.  SPACES.
+# SCREW THIS UP AND IT WON'T WORK.
+# Also, all entries are changed into a single [] after a master changelog generation. Just remove the brackets when you add new entries.
+# Please surround your changes in  double quotes ("), as certain characters otherwise screws up compiling. The quotes will not show up in the changelog.
+changes: 
+  - rscadd: "Distress Beacons are here! They replace the standard ERT call with a cool sound effect."
+  - rscadd: "Activating a distress beacon is the same as always. Once activated, a random ERT will be selected between, for now, independent neutral mercenaries, TCFL and Nanotrasen ERT."
+  - rscadd: "Added a cool new ERT base for outsider ERTs to the centcomm level, courtesy of Fernerr."
+  - rscdel: "Removed the Join Response Team verb. All ERTs are now ghost spawners."
+  - rscadd: "More ERTs to come. Unathi Kataphracts soon! They will not be whitelisted."
```
[the remark this answers is not in the document]
Comment (NonQueueingMatt) [reply html/changelogs/mattatlas-mybloodpacttoskull.yml]: addressed

File: code/controllers/subsystems/responseteam.dm [hunk 8]
```diff
@@ -2,123 +2,96 @@
...
+	command_announcement.Announce("A distress beacon has been launched. Please remain calm, a relief team will arrive soon.", "[current_map.boss_name]", 'sound/effects/distressbeacon.ogg')
 
-	if(forced_choice)
-		ert_type = forced_choice
-	else if(!forced_choice && !prob(ert_chance)) 
-		ert_type = "TCFL"
+	if(forced_choice && forced_choice != "Random")
+		for(var/datum/responseteam/R in available_teams)
+			if(R.name == forced_choice)
+				picked_team = R
```
[the remark this answers is not in the document]
Comment (NonQueueingMatt) [reply code/controllers/subsystems/responseteam.dm]: Solved

File: code/controllers/subsystems/responseteam.dm [hunk 9]
```diff
@@ -2,123 +2,96 @@
...
+	var/list/all_teams = subtypesof(/datum/responseteam)
+	if(!all_teams)
+		log_debug("No response teams found!")
+		return
+	else
+		for(var/team in all_teams)
+			CHECK_TICK
+			var/datum/responseteam/ert = new team
+			if(!ert) 
+				continue
```
[the remark this answers is not in the document]
Comment (NonQueueingMatt) [reply code/controllers/subsystems/responseteam.dm]: Solved

File: code/modules/ghostroles/spawner/human/ert.dm [hunk 10]
```diff
@@ -0,0 +1,102 @@
...
+	max_count = 2
+	desc = "A medic of the Tau Ceti Foreign Legion."
+	outfit = /datum/outfit/admin/ert/legion/specialist
+	mob_name_prefix = "Lgn."
+
+/datum/ghostspawner/human/ert/tcfl/leader
+	name = "TCFL Leader"
+	short_name = "tcfll"
+	max_count = 1
+	mob_name_prefix = "Pfct."
```
Comment (fernerr) [inline code/modules/ghostroles/spawner/human/ert.dm]:
```suggestion
	mob_name_prefix = "Pfct. "
```

File: code/modules/ghostroles/spawner/human/ert.dm [hunk 11]
```diff
@@ -0,0 +1,102 @@
...
+	spawnpoints = list("TCFLERTSpawn")
+	possible_species = list("Human", "Tajara", "Skrell", "Unathi", "Vaurca Warrior", "Vaurca Worker", "Machine")
+
+/datum/ghostspawner/human/ert/tcfl/specialist
+	name = "TCFL Medical Specialist"
+	short_name = "tcfls"
+	max_count = 2
+	desc = "A medic of the Tau Ceti Foreign Legion."
+	outfit = /datum/outfit/admin/ert/legion/specialist
+	mob_name_prefix = "Lgn."
```
Comment (fernerr) [inline code/modules/ghostroles/spawner/human/ert.dm]:
```suggestion
	mob_name_prefix = "Lgn. "
```

**Comment by NonQueueingMatt** [comment]
I'm leaving the outfit names as is for ease of access.

**Comment by NonQueueingMatt** [comment]
nvm lmao

File: code/modules/ghostroles/spawner/human/ert.dm [hunk 12]
```diff
@@ -0,0 +1,102 @@
...
+	desc = "The leader of the Nanotrasen Phoenix ERT."
+	max_count = 1
+	mob_name_prefix = "L/Tpr. "
+
+//TCFL ERT
+/datum/ghostspawner/human/ert/tcfl
+	name = "TCFL Responder"
+	short_name = "tcflr"
+	desc = "The Tau Ceti Foreign Legion's rank and file."
+	welcome_message = "The Tau Ceti Foreign Legion works for the Republic of Biesel; your job is to protect the place you're heading to and fix the problem. You can be a volounteer (Vol.) or a legionnaire (Lgn.). The former is reccomended for new players."
```
Comment (fernerr) [inline code/modules/ghostroles/spawner/human/ert.dm]:
```suggestion
	welcome_message = "The Tau Ceti Foreign Legion works for the Republic of Biesel; your job is to protect the place you're heading to and fix the problem. You can be a volounteer (Vol.) or a legionnaire (Lgn.). The former is recommended for new players."
```

File: code/controllers/subsystems/responseteam.dm [hunk 13]
```diff
@@ -2,123 +2,96 @@
...
 	NEW_SS_GLOBAL(SSresponseteam)
 	feedback_set("responseteam_count",0)
 
+/datum/controller/subsystem/responseteam/Initialize(start_timeofday)
+	. = ..()
+	var/list/all_teams = subtypesof(/datum/responseteam)
+	if(!all_teams)
+		log_debug("No response teams found!")
+		return
+	else
```
Comment (skull132) [inline code/controllers/subsystems/responseteam.dm]: this else is unnecessary.

Comment (NonQueueingMatt) [reply code/controllers/subsystems/responseteam.dm]: Shit, how could you do this to me? I trusted you, Skull. How dare you ruin my life like this. How dare you shit on my code like this. This else was the product of 3 hours of consecutive thinking. And you just came in here like this. Why would you do this, Skull?

File: code/controllers/subsystems/responseteam.dm [hunk 14]
```diff
@@ -2,123 +2,92 @@
...
-				progression_chance += config.ert_blue_inc
-				pg_red += config.ert_red_inc
-			if("delta")
-				progression_chance += config.ert_delta_inc
-				pg_delta += config.ert_delta_inc
-			else
-				progression_chance += 1
-
-/datum/controller/subsystem/responseteam/proc/trigger_armed_response_team(var/forced_choice = FALSE)
-	if(!can_call_ert && !forced_choice)
```
Comment (Arrow768) [inline code/controllers/subsystems/responseteam.dm]: By removing `&& !forced_choice` you removed the ability for admins to force a second specific response team (and automatically alert the crew with the default announcement)

File: code/controllers/subsystems/responseteam.dm [hunk 15]
```diff
@@ -2,123 +2,92 @@
...
+	for(var/datum/ghostspawner/G in sent_teams)
+		G.disable()
+
+/datum/controller/subsystem/responseteam/proc/handle_spawner()
+	for(var/N in typesof(picked_team.spawner)) //Find all spawners that are subtypes of the team we want.
+		var/datum/ghostspawner/human/ert/new_spawner = new N
+		for(var/role_spawner in SSghostroles.spawners)
+			if(new_spawner.short_name == role_spawner) //Create the spawner, then use its name to find the spawner in SSghostroles' spawner lists.
+				var/datum/ghostspawner/human/ert/good_spawner = SSghostroles.spawners[role_spawner]
+				sent_teams += good_spawner //Enable that spawner.
```
Comment (Arrow768) [inline code/controllers/subsystems/responseteam.dm]: Ensure to clear sent teams before using it in this proc (and after verifying that another team isnt in the process of being sent)

File: code/datums/ert/responseteam.dm [hunk 16]
```diff
@@ -0,0 +1,5 @@
+/datum/responseteam
+	var/name = "Default Team"					//ERT name.
```
Comment (Arrow768) [inline code/datums/ert/responseteam.dm]: Use space for the comment indentation

File: code/game/objects/items/weapons/cards_ids.dm [hunk 17]
```diff
@@ -404,12 +404,18 @@ var/const/NO_EMAG_ACT = -50
...
-/obj/item/weapon/card/id/legion
-	name = "\improper Tau Ceti Foreign Legion ID"
+/obj/item/weapon/card/id/distress
+	name = "\improper Freelancer Mercenary ID"
 	icon_state = "centcom"
-	overlay_state = "centcom"
+	assignment = "Freelancer Mercenary"
+
+/obj/item/weapon/card/id/distress/New()
+	access = list(access_legion, access_distress, access_maint_tunnels, access_external_airlocks, access_security, access_engine, access_engine_equip, access_medical, access_research, access_atmospherics, access_medical_equip)
```
Comment (Arrow768) [inline code/game/objects/items/weapons/cards_ids.dm]: That is something I am unsure about for the Freelance Mercs.

If they are not associated with NT in any way, then why would have have access to a NT installation.

Comment (NonQueueingMatt) [reply code/game/objects/items/weapons/cards_ids.dm]: I mean this is legion tier access. It's the bare minimum to get around maint.

Comment (Arrow768) [reply code/game/objects/items/weapons/cards_ids.dm]: Well yes, but the legion is somewhat in bed with NT.
These mercs are neutral so they wouldnt (and shouldnt) have any access.

If they indeed start working for NT (which is not a given) then command can hand out access as needed.

File: code/controllers/subsystems/responseteam.dm (hunk 14 above)
Comment (NonQueueingMatt) [reply code/controllers/subsystems/responseteam.dm]: Solved

File: code/controllers/subsystems/responseteam.dm (hunk 15 above)
Comment (NonQueueingMatt) [reply code/controllers/subsystems/responseteam.dm]: Solved

File: code/datums/ert/responseteam.dm (hunk 16 above)
Comment (NonQueueingMatt) [reply code/datums/ert/responseteam.dm]: Solved

File: code/game/objects/items/weapons/cards_ids.dm (hunk 17 above)
Comment (NonQueueingMatt) [reply code/game/objects/items/weapons/cards_ids.dm]: Solved

**Pull request merged**